homeassistant/components/zwave_js/entity.py

```diff
@@ -121,25 +121,45 @@ def get_zwave_value(
         endpoint: Optional[int] = None,
         value_property_key_name: Optional[str] = None,
         add_to_watched_value_ids: bool = True,
+        check_all_endpoints: bool = False,
     ) -> Optional[ZwaveValue]:
         """Return specific ZwaveValue on this ZwaveNode."""
         # use commandclass and endpoint from primary value if omitted
         return_value = None
         if command_class is None:
             command_class = self.info.primary_value.command_class
-        if endpoint is None:
-            endpoint = self.info.primary_value.endpoint
+
+        # Build partial event data dictionary so we can change the endpoint later
+        partial_evt_data = {
+            "commandClass": command_class,
+            "property": value_property,
+            "propertyKeyName": value_property_key_name,
+        }
+
         # lookup value by value_id
         value_id = get_value_id(
             self.info.node,
             {
-                "commandClass": command_class,
-                "endpoint": endpoint,
-                "property": value_property,
-                "propertyKeyName": value_property_key_name,
+                **partial_evt_data,
+                "endpoint": endpoint or self.info.primary_value.endpoint,
             },
         )
         return_value = self.info.node.values.get(value_id)
+
+        # If we haven't found a value, the endpoint was not provided, and we want to
+        # search for the value on additional endpoints, return the next value that
+        # can be found, if any
+        if return_value == endpoint is None and check_all_endpoints:
```

Comment:
This check syntax is weird.

Reply:
Is this the weird part `return_value == endpoint is None`? I can change this to `return_value is None and endpoint is None`. Do we need to check if `endpoint is None` here? I suppose if someone makes the flag true, we should check other endpoints even if they specified one in the call